core/src/main/java/org/elasticsearch/ingest/processor/SplitProcessor.java

```diff
@@ -60,7 +63,9 @@ public void execute(IngestDocument document) {
         if (oldVal == null) {
             throw new IllegalArgumentException("field [" + field + "] is null, cannot split.");
         }
-        document.setFieldValue(field, Arrays.asList(oldVal.split(separator)));
+        List<String> splitList = new ArrayList<>();
```

Comment:
maybe `List<String> splitList = new ArrayList<>(oldVal.split(separator));` that way we use the right size for that array list?